base/gmp.jl

```diff
@@ -104,7 +111,7 @@ function tryparse_internal(::Type{BigInt}, s::AbstractString, startpos::Int, end
         raise && throw(ArgumentError("invalid BigInt: $(repr(bstr))"))
         return _n
     end
-    Nullable(sgn < 0 ? -z : z)
+    Nullable(flipsign!(z, sgn))
```

Comment:
As `z` was created in the function, it "owns" this object is hence is free to mutate it for optimization, which is the main purpose of this PR: mutating `BigInt` objects in place when possible to avoid expensive copies.